spec/controllers/volunteers_controller_spec.rb

```diff
@@ -2,8 +2,320 @@
+
+    before { sign_in admin }
+
+    it 'renders the index template' do
+      expect(subject).to render_template :index
+    end
+  end
+
+  describe 'GET #edit' do
+    subject do
+      get :edit, { id: volunteer.id }
+    end
+
+    before { sign_in volunteer }
+
+    it 'renders the edit template' do
+      expect(subject).to render_template :edit
+    end
+  end
+
+  describe 'PUT #update' do
+    let(:name) { 'Mister Mxyzptlk' }
+    let(:valid_params) do
+      {
+        id: volunteer.id,
+        volunteer: {
+          name: name
+        }
+      }
+    end
+
+    subject { put :update, valid_params }
+
+    before { sign_in volunteer }
+
+    it 'renders the index template' do
+      expect(subject).to render_template :index
+    end
+  end
+
+  describe 'GET #switch_user' do
+    subject do
+      get :switch_user, { volunteer_id: volunteer.id }
+    end
+
+    context 'non-admin user' do
+      before { sign_in volunteer }
+
+      it 'disallows the action' do
+        expect(subject).to redirect_to root_path
+      end
+    end
+
+    context 'admin user' do
+      before do
+        volunteer.update_attribute(:waiver_signed, true)
+        sign_in admin
+      end
+
+      it 'renders the home template' do
+        expect(subject).to render_template :home
+      end
+    end
+  end
+
+  describe 'GET #region_admin' do
+    subject { get :region_admin }
+
+    before { sign_in volunteer }
+
+    it 'renders the region_admin template' do
+      expect(subject).to render_template :region_admin
+    end
+  end
+
+  describe 'GET #stats' do
+    subject { get :stats }
+
+    context 'non-admin user' do
+      before { sign_in volunteer }
+
+      it 'disallows the action' do
+        expect(subject).to redirect_to root_path
+      end
+    end
+
+    context 'admin user' do
+      before { sign_in admin }
+
+      it 'renders the stats template' do
+        expect(subject).to render_template :stats
+      end
+    end
+  end
+
+  describe 'GET #waiver' do
+    subject { get :waiver }
+
+    before { sign_in volunteer }
+
+    it 'renders the waiver template' do
+      expect(subject).to render_template :waiver
+    end
+  end
+
+  describe 'GET #sign_waiver' do
+    subject do
+      get :sign_waiver, { accept: '1' }
+    end
+
+    before { sign_in volunteer }
+
+    it 'renders the home template' do
+      expect(subject).to render_template :home
+    end
+  end
+
+  describe 'GET #knight' do
+    subject do
+      get :knight, { volunteer_id: volunteer.id }
+    end
+
+    context 'non-admin user' do
+      before { sign_in volunteer }
+
+      it 'disallows the action' do
+        expect(subject).to redirect_to root_path
+      end
+    end
+
+    context 'admin user' do
+      before { sign_in admin }
+
+      it 'renders the knight template' do
+        # TODO: fails with undef ar or meth `admin` in controller.
+        # Should be `admin_only`? Which is not a route. Or, `admin_only`
+        # should be `admin`?
```

Comment:
Agreed. I will make it a pending spec once it is merged and I pull down the base repo.